doc/pveclibmaindox.h

```diff
@@ -1104,59 +1215,1050 @@ s0 = vec_addeq (&c0, a0, b0, c1);
+*
+* \subsubsection main_libary_issues_0_0_1_1 Static linkage to platform specific functions
+* For static linkage the application is compiled for a specific
+* platform target (via -mcpu=). So function calls should be bound to
+* the matching platform specific implementations. The application
+* may select the platform specific function directly by defining
+* a <I>extern</I> and invoking the platform qualified function.
+*
+* Or simply use the __VEC_PWR_IMP() macro as wrapper for the function
+* name in the application.
+* This selects the appropriate platform specific implementation based
+* on the -mcpu= specified for the application compile.
+* For example.
+* \code
+  k = __VEC_PWR_IMP (vec_mul128x128)(i, j);
+* \endcode
+*
+* The vec_int512_ppc.h header provides the default platform qualified
+* <I>extern</I>  declarations for this and related functions based on
+* the \-mcpu= specified for the compile of application including this
+* header. For example.
+* \code
+extern __VEC_U_256
+__VEC_PWR_IMP (vec_mul128x128) (vui128_t, vui128_t);
+* \endcode
+* For example if the applications calling vec_mul128x128() is itself
+* compiled with -mcpu=power8, then the __VEC_PWR_IMP() will insure
+* that:
+* - The vec_int512_ppc.h header will define an extern for
+* vec_mul128x128_PWR8.
+* - That application's calls to __VEC_PWR_IMP (vec_mul128x128) will
+* reference vec_mul128x128_PWR8.
+*
+* The application should then link to the libpvecstatic.a archive.
+* Where the application references PVECLIB functions with the
+* appropriate target suffix, the linker will extract only the matching
+* implementations and include them in the program image.
+*
+* \subsubsection main_libary_issues_0_0_1_2 Dynamic linkage to platform specific functions
+* Applications using dynamic linkage will call the unqualified
+* function symbol.
+* For example:
+* \code
+extern __VEC_U_256
+vec_mul128x128 (vui128_t, vui128_t);
+* \endcode
+*
+* This symbol's implementation (in libpvec.so) has a special
+* <B>STT_GNU_IFUNC</B>
+* attribute recognized by the dynamic linker which associates this
+* symbol with the corresponding runtime resolver function.
+* The application simply calls the (unqualified) function and the
+* dynamic linker (with the help of PVECLIB's IFUNC resolvers) handles
+* the details.
+*
+* On the application first call to a <I>IFUNC</I> symbol, the dynamic
+* linker calls the resolver function associated with that symbol.
+* The resolver function performs a runtime check to determine the platform,
+* selects the (closest) matching platform specific function,
+* then returns that functions address to the dynamic linker.
+*
+* The dynamic linker stores this function address in the callers
+* Procedure Linkage Tables (PLT) before forwarding the call to the
+* resolved implementation.  Any subsequent calls to this
+* function symbol branch (via the PLT) directly to appropriate
+* platform specific implementation.
```

Comment:
Didn't we already cover this near line 2111? https://github.com/open-power-sdk/pveclib/pull/100/files#diff-65c017f96ba9d7c80101e9a2263e4ed7R2111

Reply:
yes deleting it